runtime/src/accounts_db.rs

```diff
@@ -1944,6 +1944,7 @@ impl AccountsDb {
                     .map(|store_removed| store_removed.contains(&account_info.offset))
                     .unwrap_or(false);
                 if was_reclaimed {
+                    *ref_count -= 1; // since we aren't including this account info, we also have to decrement the expected refcount so the list len matches the expected ref count
```

Comment:
hmmm I don't think you want to decrement the ref count here because the `was_reclaimed` means the entry for this `(pubkey, slot)` was removed from the AccountsIndex, but still exists in the storage entry until a shrink happens